src/zoid/buttons/component.jsx

```diff
@@ -681,6 +682,84 @@ export const getButtonsComponent: () => ButtonsComponent = memoize(() => {
         },
       },
 
+      onMessageHover: {
+        type: "function",
+        required: false,
+        value: ({ props, state }) => {
+          return async ({ offerType }) => {
+            if (!window.paypal.MessagesModal.show && !state.isModalFetching) {
+              state.isModalFetching = true;
+              await getModal();
+              state.isModalFetching = false;
+
+              if (state.isAwaitingShow === true) {
+                const { message, clientID, merchantID } = props;
+                const amount = message?.amount;
+
+                window.paypal
+                  .MessagesModal({
+                    onReady: ({ show }) => {
+                      show({
+                        amount,
+                        offer: offerType,
+                        account: clientID,
+                        merchantId: merchantID,
+                      });
+                    },
+                  })
+                  .render("body");
+
+                state.isAwaitingShow = false;
```

Comment:
this block is written for an edge case where a user has hovered, thus starting the modal download, and then clicked before the download has finished.  To do: make sure that `state` updates as I believe it does, and make sure click logging occurs in this case.

Reply:
removed this for a better solution